runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/FusedPipeline.java

```diff
@@ -36,25 +36,45 @@
 @AutoValue
 public abstract class FusedPipeline {
   static FusedPipeline of(
-      Set<ExecutableStage> environmentalStages, Set<PTransformNode> runnerStages) {
-    return new AutoValue_FusedPipeline(environmentalStages, runnerStages);
+      Components components,
+      Set<ExecutableStage> environmentalStages,
+      Set<PTransformNode> runnerStages) {
+    return new AutoValue_FusedPipeline(components, environmentalStages, runnerStages);
   }
 
+  abstract Components getComponents();
+
   /** The {@link ExecutableStage executable stages} that are executed by SDK harnesses. */
   public abstract Set<ExecutableStage> getFusedStages();
 
   /** The {@link PTransform PTransforms} that a runner is responsible for executing. */
   public abstract Set<PTransformNode> getRunnerExecutedTransforms();
 
-  public RunnerApi.Pipeline toPipeline(Components initialComponents) {
-    Map<String, PTransform> executableTransforms = getExecutableTransforms(initialComponents);
-    Components fusedComponents = initialComponents.toBuilder()
-        .putAllTransforms(executableTransforms)
-        .putAllTransforms(getFusedTransforms())
-        .build();
+  /**
+   * Returns the {@link RunnerApi.Pipeline} representation of this {@link FusedPipeline}.
+   *
+   * <p>The {@link Components} of the returned pipeline will contain all of the {@link PTransform
+   * PTransforms} present in the original Pipeline that this {@link FusedPipeline} was created from,
+   * plus all of the {@link ExecutableStage ExecutableStages} contained within this {@link
+   * FusedPipeline}. The Root Transform IDs will contain all of the runner executed transforms and
```

Comment:
The upper casing on Root Transform IDs is strange, would you rather link the Pipeline root transform ids method?